x-team/xp-ui
# Add custom error message for legacy devices

…s error in older browsers



**Release Type:** *Bug Fix* 


Fixes https://x-team-internal.atlassian.net/browse/XP-3643

## Description

It suggests the user use the latest browser version for errors caused in the following older browsers when they do not support `RequestAPI` and `URLSearchParams`:
- Chrome v48-
- Safari v9.1-
- Firefox v38-

## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Steps to Test or Reproduce

In `xp-ui`:
1. Run `rm -r node_modules && npm i`
2. Run `npm run start` and open `http://localhost:9001/`
3. Scroll to the section "Core Components" > "ErrorBox" > "Debug" > "Custom error message for legacy devices"
4. Confirm the error displayed is "Please update your browser to the latest version" for the following points:
    - 'error \'TypeError: invalid \'instanceof\' operand y.Request\''
    - 'error \'TypeError: Expecting a function in instanceof check, but got undefined\''
    - 'error \'TypeError: y.Request is not a function. (evaluating \'t instanceof y.Request\')\''
    - 'error \'ReferenceError: URLSearchParams is not defined\''

## Impacted Areas in Application

- `ErrorBox.js`

## Screenshots

- 'error \'TypeError: invalid \'instanceof\' operand y.Request\''

![Screen Shot on 2020-04-10 at 15:30:36](https://user-images.githubusercontent.com/6010411/79017924-805e2b80-7b40-11ea-9268-e3e89db4c4af.png)

- 'error \'TypeError: Expecting a function in instanceof check, but got undefined\''

![Screen Shot on 2020-04-10 at 15:30:49](https://user-images.githubusercontent.com/6010411/79017946-90760b00-7b40-11ea-96e3-6635e98c7550.png)

- 'error \'TypeError: y.Request is not a function. (evaluating \'t instanceof y.Request\')\''

![Screen Shot on 2020-04-10 at 15:30:59](https://user-images.githubusercontent.com/6010411/79017953-94a22880-7b40-11ea-8b46-b9f1b744e1c8.png)

- 'error \'ReferenceError: URLSearchParams is not defined\''

![Screen Shot on 2020-04-10 at 15:31:09](https://user-images.githubusercontent.com/6010411/79017963-9966dc80-7b40-11ea-9d90-f06796c584ba.png)
